### PR TITLE
Start & stop events fixes

### PR DESCRIPTION
**Pull request by cianclarke**
* Add start & stop event reporting in node.js rather than control scripts. 
* DRY up reporting agent logic
* Implement tidy

### CODE REVIEW

**Comment by cianclarke** [comment]
@jasonmadigan @david-martin review on this?

File: lib/notifications.js [hunk 1]
```diff
@@ -9,3 +14,50 @@ exports.create = function(notification, cb){
...
+      }  
+      pid = parseInt(fileContents.toString(), 10);
+      if (isNaN(pid)){
+        // Something gone wrong - retry in a bit.
+        return;
+      }
+      // We've successfully started
+      logger.info('Node.js cartridge start detected');
+      clearInterval(interval);
+      notifications.push({ type : "DYNOMAN_APP_START_SUCCEEDED" });
```
Comment (david-martin) [inline lib/notifications.js]: why DYNOMAN?

**Comment by david-martin** [comment]
reviewed

File: lib/notifications.js (hunk 1 above)
Comment (cianclarke) [inline lib/notifications.js]: `DYNOMAN_APP_START_SUCCEEDED` becomes `fh.events.dynoman.app.start-succeeded` here: https://github.com/fheng/fh-amqp-js/blob/master/lib/fhevents.js
which becomes some other things somewhere in here https://github.com/fheng/millicore/blob/a400c2969369b94c8719b9fcbf1d569e5a20b841/src/main/java/org/tssg/millicore/eventlog/EventMessagePublisher.java that I didn't really want to tackle :)

Comment (david-martin) [inline lib/notifications.js]: ok. should add a jira (low priority) for sorting this out

Comment (cianclarke) [inline lib/notifications.js]: Good idea https://issues.jboss.org/browse/FHMAP-172

**Pull request merged**